### PR TITLE
added ability to add attributes to datamodels

### PR DESCRIPTION
**Pull request by perrygreenfield**
Added the ability to add attributes to datamodels (using the same mechanism as for TaggedObjectNodes, i.e., by using dictionary indices).

Closes [#32]

### CODE REVIEW

File: tests/test_models.py [hunk 1]
```diff
@@ -201,18 +206,30 @@ def test_opening_mask_ref(tmp_path):
...
 def test_opening_readnoise_ref(tmp_path):
     # First make test reference file
     file_path = tmp_path / 'testreadnoise.asdf'
     utils.mk_readnoise(filepath=file_path)
     readnoise = datamodels.open(file_path)
     assert readnoise.meta.instrument.optical_element == 'F158'
     assert isinstance(readnoise, datamodels.ReadnoiseRefModel)
+
+
+def test_add_model_attribute(tmp_path):
```
Comment (eslavich) [inline tests/test_models.py:227]: Do we need to be able to save these new attributes to the .asdf file, and read them back again?  That's worth a test, if so.

**Review by PaulHuwe — COMMENTED** [review]
LGTM

File: src/roman_datamodels/datamodels.py [hunk 2]
```diff
@@ -179,6 +180,12 @@ def __setattr__(self, attr, value):
     def __getattr__(self, attr):
         return getattr(self._instance, attr)
 
+    def __setitem__(self, key, value):
```
Comment (eslavich) [inline src/roman_datamodels/datamodels.py:183]: What happens if the user sets a key from the schema using the bracket operator?  Does this bypass validation?

Also, can the new attribute be set with dot notation after it's been created?

File: tests/test_models.py (hunk 1 above)
Comment (perrygreenfield) [reply tests/test_models.py:227]: Sure, though I don't see why that won't work. But I'll add a test.

File: src/roman_datamodels/datamodels.py (hunk 2 above)
Comment (perrygreenfield) [reply src/roman_datamodels/datamodels.py:183]: I'll check the first, probably it does, so I should see if the attribute exists first. Yes, it should be modifiable by dot notation, but also a good thing to test.

File: src/roman_datamodels/datamodels.py [hunk 3]
```diff
@@ -179,6 +180,16 @@ def __setattr__(self, attr, value):
...
+        if key.startswith('_'):
+            raise ValueError(
+                'May not specify attributes/keys that start with _')
+        if hasattr(self._instance, key):
+            print('XXXXXX')
```
Comment (eslavich) [inline src/roman_datamodels/datamodels.py:188]: Sir, I caught this print statement nosing around the methods.  Do you want me to send it to the brig?

**Review by eslavich — APPROVED** [review]
One rogue line of code, but otherwise LGTM

File: src/roman_datamodels/datamodels.py (hunk 3 above)
Comment (perrygreenfield) [reply src/roman_datamodels/datamodels.py:188]: Damn, I'm so imperfect. No, I'll delete it.

**Pull request merged**